### PR TITLE
machine-api-operator: add workflow-based e2e-upgrade test for vsphere

### PR DESCRIPTION
**Pull request by vrutkovs**
Create necessary steps to install a vSphere cluster from `stable-initial:installer` and upgrade it using workflow approach.

Ref: https://issues.redhat.com/browse/OTA-308

TODO:
* [x] Requires https://github.com/openshift/origin/pull/25900

/cc @wking @stevekuznetsov @jcpowermac

### CODE REVIEW

**Comment by jcpowermac** [comment]
/lgtm
Cluster installed and upgraded as expected

**Review by michaelgugino — CHANGES_REQUESTED** [review]
/hold

We're not enabling automated vsphere testing on our repos until we have a reproducible environment.

**Comment by jcpowermac** [comment]
@vrutkovs thanks for doing this, any chance we move this from MAO to installer? If so I will approve after rebase ;-)

**Comment by vrutkovs** [comment]
>any chance we move this from MAO to installer?

Installer would use a different workflow (it doesn't need to start at "stable-initial:installer"), does it even need upgrade testing?

**Comment by jcpowermac** [comment]
> > any chance we move this from MAO to installer?
> 
> Installer would use a different workflow (it doesn't need to start at "stable-initial:installer"), does it even need upgrade testing?

Good point, how does the other platforms run an upgrade and then run tests? Or is it a specific component?

**Comment by vrutkovs** [comment]
>Good point, how does the other platforms run an upgrade and then run tests?

Correct. Other components are installing "stable-initial" imagestream (latest nightly basically) and update to the release + proposed change. This would not apply to installer and I'm not sure if it makes sense to add an update test for installer at all

**Comment by wking** [comment]
> I'm not sure if it makes sense to add an update test for installer at all

@abhinavdahiya 's previous motivation for installer update CI was something about suffering in solidarity with the rest of the org when update CI is flaking out 😉.  I'm fine dropping it.

**Comment by vrutkovs** [comment]
Failed to bootstrap

/retest

**Comment by JoelSpeed** [comment]
> We're not enabling automated vsphere testing on our repos until we have a reproducible environment.

@michaelgugino Just coming across this issue and am a bit confused why we want to hold this. IMO having some testing seems better than no testing. I'm not sure what you mean by reproducible environment here, as far as I was aware we could create vsphere clusters manually in the same way they're created in CI for manual testing, is that not that case?

**Comment by vrutkovs** [comment]
Tests are now passing in master branch, PTAL

**Comment by jcpowermac** [comment]
@vrutkovs awesome!
LGTM
will let others respond

**Review by JoelSpeed — COMMENTED** [review]
Could we confirm whether this will be blocking PRs, I was looking for `optional: true` on the presubmit but not sure if that's the correct syntax? I couldn't see it at least

**Comment by vrutkovs** [comment]
Right, I'll make them optional (but running always) for a start.

File: ci-operator/jobs/openshift/machine-api-operator/openshift-machine-api-operator-master-presubmits.yaml
Comment (JoelSpeed) [inline ci-operator/jobs/openshift/machine-api-operator/openshift-machine-api-operator-master-presubmits.yaml:215]: I think you've added this to the wrong one, this is for the AWS upgrade right?

Comment (vrutkovs) [reply ci-operator/jobs/openshift/machine-api-operator/openshift-machine-api-operator-master-presubmits.yaml]: good catch, fixed

**Comment by vrutkovs** [comment]
/retest

Install error

**Review by JoelSpeed — COMMENTED** [review]
/lgtm

**Comment by vrutkovs** [comment]
/hold cancel

lets merge it and adjust amount of periodics based on performance

**Pull request merged**
